rules/singular.mutations.js

```diff
@@ -19,12 +19,43 @@ const getMessage = (text, name, start) => {
   };
 };
 
+function getMutationNames(ast) {
+  const mutationNames = [];
+
+  visit(ast, {
+    ObjectTypeExtension(node) {
+      if (
+        node.name.value !== 'Mutation' ||
+        node.name.value === 'GraphQLMutation'
+      ) {
+        return;
+      }
+      visit(node, {
+        FieldDefinition(node) {
+          mutationNames.push(node.type.name.value);
+        }
+      });
+    },
+    ObjectTypeDefinition(node) {
+      if (
+        node.name.value === 'Mutation' ||
+        node.name.value === 'GraphQLMutation'
+      ) {
+        mutationNames.push(node.name.value);
+      }
+    }
+  });
+  return mutationNames;
+}
+
 module.exports = function(ast, text) {
   const messages = [];
 
+  const mutationNames = getMutationNames(ast);
+
   visit(ast, {
     ObjectTypeDefinition(node) {
-      if (node.name.value !== 'GraphQLMutation') {
+      if (mutationNames.indexOf(node.name.value) < 0) {
```

Comment:
And lastly, would you be able to add a test for this new "extend type Mutation" in https://github.com/happylinks/gqlint/blob/master/tests/rules/removedelete.mutations.test.js and maybe also in https://github.com/happylinks/gqlint/blob/master/tests/rules/singular.mutations.test.js just to be sure.